### PR TITLE
Patch CVE-2023-31486 in perl

### PR DESCRIPTION
**Pull request by corvus-callidus**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This patches CVE-2023-31486 for perl. This patch changes the default behavior of the embedded HTTP-Tiny package to verify certificates by default and adds support for an environment variable to switch back to pre-patch behavior (don't verify by default). The patch for HTTP-Tiny was created for v0.0.86 and included extra tests and documentation changes. These extras were omitted from this patch. Additionally, a subsequent commit to HTTP-Tiny fixed a defect in the original patch. The relevent portion of that fix is also included here.

###### Change Log  

- Patches CVE-2023-31486 for perl-5.30.3

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-31486

###### Test Methodology

- Pipeline builds: 
  - [ARM64 BuildToolchain](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=387703&view=results) - PASSED
  - [ARM64 BuildRPMS](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=387744&view=results) - PASSED
  - [AMD64 BuildToolchain](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=388328&view=results) - PASSED
  - [AMD64 BuildRPMS](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=388363&view=results) - PASSED

### CODE REVIEW

**Comment by SeanDougherty** [comment]
LGTM!

**Review by rmhsawyer — APPROVED** [review]

**Review by SeanDougherty — APPROVED** [review]

File: SPECS/perl/perl.spec [hunk 1]
```diff
@@ -34,6 +35,7 @@ The Perl package contains the Practical Extraction and
 Report Language.
 %prep
 %setup -q
```
Comment (henryli001) [inline SPECS/perl/perl.spec:37]: Suggests changing to %autosetup -p1

File: SPECS/perl/perl.spec [hunk 2]
```diff
@@ -74,6 +76,10 @@ make test TEST_SKIP_VERSION_CHECK=1
 %{_mandir}/*/*
 
 %changelog
+* Fri Jun 30 2023 corvus-callidus <108946721+corvus-callidus@users.noreply.github.com> - 5.30.3-3
```
Comment (neha170) [inline SPECS/perl/perl.spec:79]: nit: please see the convention for username here, thanks

File: SPECS/perl/perl.spec (hunk 1 above)
Comment (corvus-callidus) [reply SPECS/perl/perl.spec:37]: After speaking offline, I'm going to decline to change this.

**Pull request merged**